src/NuGet.Services.AzureSearch/ScoringProfiles/DefaultScoringProfile.cs

```diff
@@ -12,44 +12,47 @@ public static class DefaultScoringProfile
     {
         public const string Name = "nuget_scoring_profile";
 
-        public static readonly ScoringProfile Instance = new ScoringProfile(
-            Name,
-            textWeights: new TextWeights
-            {
-                Weights = new Dictionary<string, double>
+        public static ScoringProfile Create(AzureSearchScoringConfiguration config)
+        {
+            return new ScoringProfile(
+                Name,
+                textWeights: new TextWeights
                 {
-                    // Exact match of the package id should be boosted the highest,
-                    // followed by a tokenized match on the package id.
-                    { IndexFields.PackageId, 10 },
-                    { IndexFields.TokenizedPackageId, 5 },
-                }
-            },
-            functions: new List<ScoringFunction>()
-            {
-                // Greatly boost results with high download counts. We score off the log of the download count
-                // with linear interpolation so that the boost slows down at higher download counts. We cannot
-                // use the raw download count with a log interpolation as that would result in a large boosting
-                // range, which would need to be offset by an unmanageably high boosting factor.
-                new MagnitudeScoringFunction(
-                    fieldName: IndexFields.Search.LogBase2DownloadCount,
-                    boost: 100.0,
-                    parameters: new MagnitudeScoringParameters(
-                        boostingRangeStart: 0,
-                        boostingRangeEnd: Math.Log(999_999_999_999, 2),
-                        shouldBoostBeyondRangeByConstant: true),
-                    interpolation: ScoringFunctionInterpolation.Linear),
+                    Weights = new Dictionary<string, double>
+                    {
+                        // Exact match of the package id should be boosted the highest,
+                        // followed by a tokenized match on the package id.
+                        { IndexFields.PackageId, config.PackageIdFieldWeight },
+                        { IndexFields.TokenizedPackageId, config.TokenizedPackageIdFieldWeight },
```

Comment:
Thoughts on putting this entire dictionary in config? Not sure if you are anticipating other fields getting weights.

Reply:
That sounds good to me, it'll give us better flexibility

Reply:
Only drawback is that now the field names are set in two repos. Not sure of the behavior if the field does not exist.